var/spack/repos/builtin/packages/perl-perlio-utf8-strict/package.py

```diff
@@ -12,4 +12,7 @@ class PerlPerlioUtf8Strict(PerlPackage):
     homepage = "https://metacpan.org/pod/PerlIO::utf8_strict"
     url = "http://search.cpan.org/CPAN/authors/id/L/LE/LEONT/PerlIO-utf8_strict-0.002.tar.gz"
 
+    version("0.009", sha256="ba82cf144820655d6d4836d12dde65f8895a3d905aeb4aa0b421249f43284c14")
     version("0.002", sha256="6e3163f8a2f1d276c975f21789d7a07843586d69e3e6156ffb67ef6680ceb75f")
+
+    depends_on("perl-module-build", type="build", when="@:0.005")
```

Comment:
My recommendation before was inaccurate given what you pointed out in the Spack documentation. In order to follow the guidance in the Spack documentation, this should be present for all versions. So, this should be reverted to what you had before; sorry about that.